swift/Workflow/Tests/SubtreeManagerTests.swift

```diff
@@ -294,6 +296,71 @@ final class SubtreeManagerTests: XCTestCase {
         observer.send(value: ())
         wait(for: [notEmittedExpectation], timeout: 1.0)
     }
+
+    // MARK: - SideEffect
+
+    func test_maintainsSideEffectLifetimeBetweenRenderPasses() {
+        let manager = WorkflowNode<ParentWorkflow>.SubtreeManager()
+        XCTAssertTrue(manager.sideEffectLifetimes.isEmpty)
+
+        _ = manager.render { context -> TestViewModel in
+            context.runSideEffect(key: "helloWorld") { _ in }
+            return context.render(
+                workflow: TestWorkflow(),
+                key: "",
+                outputMap: { _ in AnyWorkflowAction.noAction }
+            )
+        }
+
+        XCTAssertEqual(manager.sideEffectLifetimes.count, 1)
+        let sideEffectKey = manager.sideEffectLifetimes.values.first!
+
+        _ = manager.render { context -> TestViewModel in
+            context.runSideEffect(key: "helloWorld") { _ in }
```

Comment:
We should probably put an `XCTFail()` in that block to ensure that the side-effect isn’t executed the second time. (As is, I think this test would pass if we didn’t correctly reuse side-effects)